### DISCUSSION TITLE
How to handle dynamically inserted layers?

### DISCUSSION THREAD

**Discussion by nardove** [uber/react-map-gl#2011] [Q&A]
My map starts with a set of pre-defined layers, the user can toggle other layers on and off, visually that is working fine and the map performs great.

But my problem is that the **newly added layer is not recognised** in the `onClick event`, I'm adding the new layers to a state I pass to `interactiveLayerIds` but this is not taking effect. Since I get the following error

```
Error: The layer 'active-lens-391' does not exist in the map's style and cannot be queried for features.
```

I've check the new source and layers are added correctly by login the event in `onSourceData` but I can't find a way to properly update the map about the newly added layer(s).

**Comment by Pessimistress** [comment]
Create a CodeSandbox if you need help.

**Reply by nardove** [reply]
I setup a basic use case [here](https://codesandbox.io/s/romantic-maria-boh7jy?file=/src/App.js) 

There is a toggle button at the top left corner, this will add a new Source and Layer component, unfortunately I don't know how to add the new added layer id to the map `interactiveLayerIds` array so I can click on its contained markers.

Hope that makes sense.

**Reply by nardove** [reply]
By using the onStyleData I manage to track down the issue, basically I was using the wrong id names, my bad, but thank you with out your prompt to create a minimal reproducible example I would not have notice the issue sooner.
